### ISSUE TITLE
Nortek_VECTOR ADV

### ISSUE THREAD

**Issue by Krys1202** [aodn/imos-toolbox#650]
Hi, 
I would like to know if the format .vec  from the Nortek Vector  could be integrated in the IMOS toolbox?
[Vector_web_LR.pdf](https://github.com/aodn/imos-toolbox/files/4259055/Vector_web_LR.pdf)
I have a raw data file " .vec"  but too big to attach here (58MB).

Cheers,
Christelle

**Comment by ocehugo** [comment]
Hi @Krys1202 ,

quick questions:
1. Is the Nortek Vector part of a new imos instrument!?
3. You may share the file in the way that works best for you (dropbox,google-drive,firefox-send,etc).
4. If applicable, may you also share a generated ASCII file from the raw data!?

**Comment by sspagnol** [comment]
For what its worth have attached a not extensively tested bit of code (was for a not IMOS related project), but it might be enough for your needs.

Please note:

- Requires slightly modified readParadoppBinary, so do all your testing in a seperate branch just in case
- I only had one instrument file example, the instrument was mounted onto a bottom frame looking down, the user has since retired and never got back to me if the resultant file looked correct. Would have to compare against the ascii exported stuff.
- Only handles ENU data.
- It doesn't handle IMU sectors.

[vectorParse.zip](https://github.com/aodn/imos-toolbox/files/4259104/vectorParse.zip)

**Comment by ocehugo** [comment]
we need to update the Paradopp help then - It says we "should" be able to read Vector files.

Problem here is that I don't have access to a file so can't do much atm

**Comment by petejan** [comment]
I have used these before, and processed them with python. 

@Krys1202 There are many options with this instrument, are you operating in burst mode? Does your vector have the AHRS installed?

**Comment by Krys1202** [comment]
> Hi @Krys1202 ,
> 
> quick questions:
> 
> 1. Is the Nortek Vector part of a new imos instrument sampling!?
> 2. You may share the file in the way that works best for you (dropbox,google-drive,firefox-send,etc).
> 3. If applicable, may you also share a generated ASCII file from the raw data!?

1-No I don't think so. We used this instrument for the AUSTEn Project (AMC/UTAS project) with Camille Couzi. If it must be an instrument used by IMOS, no worries! I just wanted to know what was possible to do :-) 
2- Link to download raw data ".vec": https://we.tl/t-uzW3gErJNM  
3- I  manage to extract .txt, but looks weird ( velocity=0)
[ADV.txt](https://github.com/aodn/imos-toolbox/files/4259298/ADV.txt)

**Comment by Krys1202** [comment]
> For what its worth have attached a not extensively tested bit of code (was for a not IMOS related project), but it might be enough for your needs.
> 
> Please note:
> 
> * Requires slightly modified readParadoppBinary, so do all your testing in a seperate branch just in case
> * I only had one instrument file example, the instrument was mounted onto a bottom frame looking down, the user has since retired and never got back to me if the resultant file looked correct. Would have to compare against the ascii exported stuff.
> * Only handles ENU data.
> * It doesn't handle IMU sectors.
> 
> [vectorParse.zip](https://github.com/aodn/imos-toolbox/files/4259104/vectorParse.zip)

Thanks for this!
 I will have a look, the configuration for this ADV is quite the same : mounted on a bottom frame with an Aquadopp, looking down .

**Comment by Krys1202** [comment]
> I have used these before, and processed them with python.
> 
> @Krys1202 There are many options with this instrument, are you operating in burst mode? Does your vector have the AHRS installed?

@petejan : Burst mode, and yes for AHRS

**Comment by petejan** [comment]
If you're willing to work with python and don't really need IMOS complaint netCDF files, then you could use,

[nortek2netCDF.py](https://github.com/petejan/imos-tools/blob/master/ocean_dp/parse/nortek2netCDF.py)

Its quite specific to what I was doing and my configuration but you might be able to make it work.

**Comment by ocehugo** [comment]
> 1-No I don't think so. We used this instrument for the AUSTEn Project (AMC/UTAS project) with Camille Couzi. If it must be an instrument used by IMOS, no worries! I just wanted to know what was possible to do :-)

It doesn't need to be an IMOS instrument - It's just that - if it's an IMOS instrument -  I need to put this in top priority.

> 2- Link to download raw data ".vec": https://we.tl/t-uzW3gErJNM
> 3- I manage to extract .txt, but looks weird ( velocity=0)
> [ADV.txt](https://github.com/aodn/imos-toolbox/files/4259298/ADV.txt)

Thanks - I will need this eventually to test and evaluate the changes required.

**Comment by sspagnol** [comment]
And from data conversion using vector software your pitch/roll values in sen file are large and lots of error messages in ssl file with "Roll out of range" and "Pitch out of range".

**Comment by Krys1202** [comment]
> If you're willing to work with python and don't really need IMOS complaint netCDF files, then you could use,
> 
> [nortek2netCDF.py](https://github.com/petejan/imos-tools/blob/master/ocean_dp/parse/nortek2netCDF.py)
> 
> Its quite specific to what I was doing and my configuration but you might be able to make it work.

Thanks for this I will have a try with python

**Comment by Krys1202** [comment]
> And from data conversion using vector software your pitch/roll values in sen file are large and lots of error messages in ssl file with "Roll out of range" and "Pitch out of range".

Yeah it doesn't seem like very good data, there was a storm during the 3-day deployment ....

**Comment by petejan** [comment]
@Krys1202 Your data file looks quite different from what I was processing, I don't see any IMU data records, so my python needs some tweaks to get it working, as I don't use the instrument pitch/roll but the data from the IMU, and did not write the system data to the output netCDF file as I did not need it for what I was looking at.
